File: test-suite/src/test/java/io/micronaut/docs/aop/introduction/IntroductionSpec.java
package io.micronaut.docs.aop.introduction;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

import io.micronaut.context.ApplicationContext;
import org.junit.Test;

public class IntroductionSpec {

    @Test
    public void testStubIntroduction() {
        try (ApplicationContext applicationContext = ApplicationContext.run()) {
            // tag::test[]
            StubExample stubExample = applicationContext.getBean(StubExample.class);

            assertEquals(10, stubExample.getNumber());
            assertNull(stubExample.getDate());
            // end::test[]
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/aop/introduction/IntroductionSpec.kt
package io.micronaut.docs.aop.introduction

import io.kotlintest.shouldBe
import io.kotlintest.specs.AnnotationSpec
import io.micronaut.context.ApplicationContext

class IntroductionSpec: AnnotationSpec() {

    @Test
    fun testStubIntroduction() {
        val applicationContext = ApplicationContext.run()

        // tag::test[]
        val stubExample = applicationContext.getBean(StubExample::class.java)

        stubExample.number.shouldBe(10)
        stubExample.date.shouldBe(null)
        // end::test[]

        applicationContext.stop()
    }
}
